Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/eventregistration/ui/EventRegistrationTestFragmentViewModel.kt

```diff
@@ -3,28 +3,131 @@ package de.rki.coronawarnapp.test.eventregistration.ui
 import androidx.lifecycle.MutableLiveData
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
-import de.rki.coronawarnapp.presencetracing.risk.CheckInRiskPerDay
-import de.rki.coronawarnapp.presencetracing.risk.CheckInWarningMatcher
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.download.DummyCheckInPackage
+import de.rki.coronawarnapp.eventregistration.checkins.split.splitByMidnightUTC
 import de.rki.coronawarnapp.presencetracing.risk.CheckInWarningOverlap
+import de.rki.coronawarnapp.presencetracing.risk.PresenceTracingRiskCalculator
+import de.rki.coronawarnapp.presencetracing.risk.launchMatching
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.debug.measureTime
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.awaitAll
+import okio.ByteString
+import org.joda.time.Instant
+import timber.log.Timber
 
 class EventRegistrationTestFragmentViewModel @AssistedInject constructor(
-    dispatcherProvider: DispatcherProvider,
-    private val checkInWarningMatcher: CheckInWarningMatcher
+    private val dispatcherProvider: DispatcherProvider,
+    private val presenceTracingRiskCalculator: PresenceTracingRiskCalculator
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
 
-    val checkInOverlaps = MutableLiveData<List<CheckInWarningOverlap>>()
-    val checkInRiskPerDayList = MutableLiveData<List<CheckInRiskPerDay>>()
+    private val checkInWarningOverlaps = mutableListOf<CheckInWarningOverlap>()
+    val checkInOverlapsText = MutableLiveData<String>()
```

Comment:
don't expose mutableLiveData..

Reply:
What is your concern? I cannot post values if is is not mutable

Reply:
ofc, you can call postValue on a MutableLiveData outside of the VM.
Just expose it as LiveData